x-pack/filebeat/docs/inputs/input-benchmark.asciidoc

```diff
@@ -0,0 +1,93 @@
+[role="xpack"]
+
+:type: benchmark
+
+[id="{beatname_lc}-input-{type}"]
+=== Benchmark input
+
+++++
+<titleabbrev>Benchmark</titleabbrev>
+++++
+
+beta[]
+
+The Benchmark input generates generic events and sends them to the output.  This can be useful when you want to benchmark the difference between outputs or output settings.
+
+Example configurations:
+
+Basic example, infinite events as quickly as possible:
+["source","yaml",subs="attributes"]
+----
+{beatname_lc}.inputs:
+- type: benchmark
+  enabled: true
+  message: "test message"
+  threads: 1
+----
+
+Send 1024 events and stop example:
+["source","yaml",subs="attributes"]
+----
+{beatname_lc}.inputs:
+- type: benchmark
+  enabled: true
+  message: "test message"
+  threads: 1
+  count: 1024
+----
+
+Send 5 events per second example:
+["source","yaml",subs="attributes"]
+----
+{beatname_lc}.inputs:
+- type: benchmark
+  enabled: true
+  message: "test message"
+  threads: 1
+  eps: 5
+----
+
+==== Configuration options
+
+The Benchmark input supports the following configuration options plus the
+<<{beatname_lc}-input-{type}-common-options>> described later.
+
+[float]
+==== `message`
+
+This is the value that will be in the `message` field of the json document.
+
+[float]
+==== `threads`
+
+This is the number of go routines that will be started generating messages.  Normally 1 thread can saturate an output but if necessary this can be increased.
```

Comment:
I believe the correct spelling is "goroutine"/"goroutines", that's how the official docs spell it, like [Effective Go](https://go.dev/doc/effective_go)

```suggestion
This is the number of goroutines that will be started generating messages.  Normally 1 thread can saturate an output but if necessary this can be increased.
```